src/mercury-common/mercury/common/configuration.py

```diff
@@ -79,3 +84,368 @@ def get_configuration(filename):
+                    f'{container} has already been assigned a value, '
+                    f'{option["name"]} is invalid')
+            namespace = namespace[container]
+        namespace[expanded_name[-1]] = option['default']
+
+    def override_with_configuration(self, option):
+        """
+
+        :param option:
+        :return:
+        """
+        value = self.format_type(
+            self.get_by_namespace(self.configuration,
+                                  option['config_address']),
+            option['special_type'])
+        if value is not None:
+            self.set_by_namespace(self.master_configuration,
+                                  option['name'],
+                                  value)
+
+    def override_with_environment(self, option):
+        """
+
+        :param option:
+        :return:
+        """
+        value = self.format_type(os.environ.get(option['env_variable']),
+                                 option['special_type'])
+        if value is not None:
+            self.set_by_namespace(self.master_configuration,
+                                  option['name'],
+                                  value)
+
+    def override_with_cli_options(self, option):
+        """
+
+        :param option:
+        :return:
+        """
+        value = self.format_type(vars(
+            self.argparse_namespace).get(
+            self.convert_argparse_destination(option['cli_argument'])),
+                                 option['special_type'])
+        if value is not None:
+            self.set_by_namespace(self.master_configuration,
+                                  option['name'],
+                                  value)
+
+    def scan_options(self):
+        """
+        :return: The master configuration
+        """
+        self.parse_args()
+        self.set_configuration()
+
+        for option in self.options:
+            # populate the namespace and fill defaults
+            self.expand_option(self.master_configuration, option)
+
+            # override defaults with configuration values
+            self.override_with_configuration(option)
+            # override defaults and config options with environment variables
+            self.override_with_environment(option)
+            # command line
+            self.override_with_cli_options(option)
+
+            # Check required
+            if option['required'] and not self.get_by_namespace(
+                    self.master_configuration, option['name']):
+                raise MercuryConfigurationError(f'Option {option["name"]} has '
+                                                f'not been provided')
+        return self.master_configuration
+
+    @staticmethod
+    def check_name(name):
+        """
+        Validator of the argument name, enforces python namespace rules
+
+        :param name: The option name
+        :return: None
+        :raises: MercuryConfigurationError
+        """
+        try:
+            ast.parse(name)
```

Comment:
Laziest check ever, I don't know if I should be impressed or disgusted.

Reply:
It works better than eveyones regex haX.. just let the interpreter do it